app/database/itemStore.go

```diff
@@ -106,21 +108,214 @@ func (s *ItemStore) GetRawNavigationData(rootID int64, userID, userLanguageID in
+			"union_table.bShowSource, " +
+			"union_table.iValidationMin, " +
+			"union_table.bShowUserInfos, " +
+			"union_table.sContestPhase, " +
+			"union_table.sUrl, " +
+			"union_table.bUsesAPI, " +
+			"union_table.bHintsAllowed, " +
+			"accessRights.fullAccess, accessRights.partialAccess, accessRights.grayedAccess, accessRights.accessSolutions "+
+			"FROM " +
+			"(SELECT items.ID AS ID, " +
+			"items.sType, " +
+			"items.bDisplayDetailsInParent, " +
+			"items.sValidationType, " +
+			"items.idItemUnlocked, " +
+			"items.iScoreMinUnlock, " +
+			"items.sTeamMode, " +
+			"items.bTeamsEditable, " +
+			"items.iTeamMaxMembers, " +
+			"items.bHasAttempts, " +
+			"items.sAccessOpenDate, " +
+			"items.sDuration, " +
+			"items.sEndContestDate, " +
+			"items.bNoScore, " +
+			"items.groupCodeEnter, " +
+			"items.bTitleBarVisible, " +
+			"items.bReadOnly, " +
+			"items.sFullScreen, " +
+			"items.bShowSource, " +
+			"items.iValidationMin, " +
+			"items.bShowUserInfos, " +
+			"items.sContestPhase, " +
+			"IF(items.sType <> 'Chapter', items.sUrl, NULL) AS sUrl, " +
+			"IF(items.sType <> 'Chapter', items.bUsesAPI, NULL) AS bUsesAPI, " +
+			"IF(items.sType <> 'Chapter', items.bHintsAllowed, NULL) AS bHintsAllowed, " +
+			"items.idDefaultLanguage, " +
+			" NULL AS iChildOrder, NULL AS sCategory, NULL AS bAlwaysVisible, NULL AS bAccessRestricted " +
+			" FROM items WHERE items.ID=? UNION ALL " +
+			"SELECT items.ID AS ID, items.sType, items.bDisplayDetailsInParent, " +
+			"items.sValidationType, items.idItemUnlocked, " +
+			"items.iScoreMinUnlock, " +
+			"items.sTeamMode, " +
+			"items.bTeamsEditable, " +
+			"items.iTeamMaxMembers, " +
+			"items.bHasAttempts, " +
+			"items.sAccessOpenDate, " +
+			"items.sDuration, " +
+			"items.sEndContestDate, " +
+			"items.bNoScore, " +
+			"items.groupCodeEnter, " +
+			"NULL AS bTitleBarVisible, " +
+			"NULL AS bReadOnly, " +
+			"NULL AS sFullScreen, " +
+			"NULL AS bShowSource, " +
+			"NULL AS iValidationMin, " +
+			"NULL AS bShowUserInfos, " +
+			"NULL AS sContestPhase, " +
+			"NULL AS sUrl, " +
+			"NULL AS bUsesAPI, " +
+			"NULL AS bHintsAllowed, " +
+			"items.idDefaultLanguage, " +
+			" iChildOrder, sCategory, bAlwaysVisible, bAccessRestricted FROM items " +
+			" JOIN items_items ON items.ID=idItemChild AND idItemParent=?) union_table " +
+			"LEFT JOIN users_items ON users_items.idItem=union_table.ID AND users_items.idUser=? " +
+			"LEFT JOIN items_strings dstrings FORCE INDEX (idItem) " +
+			" ON dstrings.idItem=union_table.ID AND dstrings.idLanguage=union_table.idDefaultLanguage " +
+			"LEFT JOIN items_strings ustrings ON ustrings.idItem=union_table.ID AND ustrings.idLanguage=? " +
+			"JOIN ? accessRights on accessRights.idItem=union_table.ID AND (fullAccess>0 OR partialAccess>0 OR grayedAccess>0) " +
+			"ORDER BY iChildOrder",
+		rootID, rootID, userID, userLanguageID, s.accessRightsSubQuery(user)).Scan(&result).Error(); err != nil {
```

Comment:
Global comment: at least the benefit of using the SQL builder is that the subqueries are easier to identify (if the variable has a proper name + comment if needed), and query parameters (?) are not 50lines further.

Reply:
I agree. As I see this, we will refactor these huge SQLs later.